handler_test.go

```diff
@@ -0,0 +1,1982 @@
+			{
+				ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+					{
+						DNSName: "www.example.com.",
+						ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+							{
+								IP:         "1.1.1.1",
+								TTLSeconds: 30,
+							},
+							{
+								IP:         "1.1.1.2",
+								TTLSeconds: 30,
+							},
+						},
+						ResolutionFailures: 0,
+						Conditions: []metav1.Condition{
+							{
+								Type:    ConditionDegraded,
+								Status:  metav1.ConditionFalse,
+								Reason:  dns.RcodeToString[dns.RcodeSuccess],
+								Message: rcodeMessage[dns.RcodeSuccess],
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		name: "Update regular dns name resolver object status with latest ttl of the corresponding IPs",
+		dnsNameResolvers: []ocpnetworkapiv1alpha1.DNSNameResolver{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "regular",
+					Namespace: "dns",
+				},
+				Spec: ocpnetworkapiv1alpha1.DNSNameResolverSpec{
+					Name: "www.example.com.",
+				},
+				Status: ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+					ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+						{
+							DNSName: "www.example.com.",
+							ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+								{
+									IP:         "1.1.1.1",
+									TTLSeconds: 20,
+								},
+								{
+									IP:         "1.1.1.2",
+									TTLSeconds: 20,
+								},
+							},
+							ResolutionFailures: 0,
+							Conditions: []metav1.Condition{
+								{
+									Type:    ConditionDegraded,
+									Status:  metav1.ConditionFalse,
+									Reason:  dns.RcodeToString[dns.RcodeSuccess],
+									Message: rcodeMessage[dns.RcodeSuccess],
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		dnsQueryTestCases: []test.Case{
+			{
+				Qname: "www.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeSuccess,
+				Answer: []dns.RR{
+					test.A("www.example.com. 30 IN A 1.1.1.1"),
+					test.A("www.example.com. 30 IN A 1.1.1.2"),
+				},
+			},
+		},
+		expectedStatuses: []ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+			{
+				ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+					{
+						DNSName: "www.example.com.",
+						ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+							{
+								IP:         "1.1.1.1",
+								TTLSeconds: 30,
+							},
+							{
+								IP:         "1.1.1.2",
+								TTLSeconds: 30,
+							},
+						},
+						ResolutionFailures: 0,
+						Conditions: []metav1.Condition{
+							{
+								Type:    ConditionDegraded,
+								Status:  metav1.ConditionFalse,
+								Reason:  dns.RcodeToString[dns.RcodeSuccess],
+								Message: rcodeMessage[dns.RcodeSuccess],
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		name: "Don't update the ttl of an IP in the regular dns name resolver object's status if next lookup time hasn't changed",
```

Comment:
The `LastLookupTime` is set to the current time when the object is created. I'll add a comment mentioning the same.

Reply:
As we are waiting for the informer to sync the various updates, if we set `LastLookupTime` to some value, by the time the test runs for this case, the next lookup time for the current queries becomes different and the test fails.

One possible solution is to have another field in the test struct to denote the time elapsed since the last lookup, which can be subtracted from the current time and used as the `LastLookupTime` when the `DNSNameResolver` object is created.

Reply:
> One possible solution is to have another field in the test struct to denote the time elapsed since the last lookup, which can be subtracted from the current time and used as the LastLookupTime when the DNSNameResolver object is created.

That seems to be a possible solution indeed. That will bring more visibility to the test cases which demonstrate the TTL behavior.